pkg/security/credential.go

```diff
@@ -92,6 +92,12 @@ func (s *Credential) getSelfCommonName() (string, error) {
 // AddSelfCommonName add Common Name in certificate that specified by s.CertPath
 // to s.CertAllowedCN
 func (s *Credential) AddSelfCommonName() error {
+	// if CertAllowedCN was not specified, we should not use any common name
+	// otherwise, https requests without a certificate will not be accepted by
+	// cdc server
+	if len(s.CertAllowedCN) == 0 {
```

Comment:
Will it be better if this condition is checked in the caller of `AddSelfCommonName`?

Reply:
yes.